### PR TITLE
Push trigrams.py to repo.

### PR DESCRIPTION
**Pull request by egpython19**

### CODE REVIEW

**Comment by egpython19** [comment]
Lesson_04 assignments complete.

**Comment by egpython19** [comment]
Mail_Room_03.py pushed to repo.

File: students/eric_gosnell/lesson_04/Mail_Room_2.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+Eric Gosnell
+Lesson 4 - Mail Room - Part 2
+11.12.2019
+"""
+
+from sys import exit
+
+# Data structure converted to dictionary
+
+DONOR_DB = {"Brent Golden": [750.00, 250.00, 250.00],
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_04/Mail_Room_2.py:11]: OK

File: students/eric_gosnell/lesson_04/Mail_Room_2.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                    "4:  Return to main menu.\n"
+                    "0:  Exit the program.\n\n"
+                    "Your choice > ")
+
+NAME_PROMPT = "Enter donor name >"
+DONATION_PROMPT = "Enter new donation amount >"
+DONOR_FOUND_MESSAGE = "{} found."
+DONOR_NOT_FOUND_MESSAGE = "{} not in database."
+DONOR_SUCCESS_MESSAGE = "{} successfully added to database."
+DONOR_EXISTS_MESSAGE = "{} already in database!  Returning to menu..."
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_04/Mail_Room_2.py:49]: Why are you using capital letters for variable names?

File: students/eric_gosnell/lesson_04/Mail_Room_2.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+def all_donors_letter():
+    """ Write multiple thank you files and print to screen - uses most recent donation """
+    for donor in DONOR_DB:
+        try:
+            with open(donor + ".txt", "w") as file:
+                file.write(THANK_YOU_MESSAGE.format(donor, DONOR_DB[donor][-1]))
+                print("\nThe following was saved to disk as a file...\n")
+                print(THANK_YOU_MESSAGE.format(donor, DONOR_DB[donor][-1]))
+        except Exception as e:
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_04/Mail_Room_2.py:118]: Nice

Comment (majstorpoki) [inline students/eric_gosnell/lesson_04/Mail_Room_2.py:118]: Which SPECIFIC exception you are trying to catch?

**Review by majstorpoki — APPROVED** [review]
Thanks

File: students/eric_gosnell/lesson_05/Mail_Room_03.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+def get_valid_name():
+    while True:
+        name = input(NAME_PROMPT)
+        try:
+            # User entered numeric data instead of name?
+            name = float(name)
+            print(INVALID_NAME)
+            continue
+        except ValueError:
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_05/Mail_Room_03.py:85]: OK

File: students/eric_gosnell/lesson_05/Mail_Room_03.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    name = get_valid_name()
+    if name in DONOR_DB:
+        print(DONOR_FOUND.format(name))
+        donation = get_valid_donation()
+        DONOR_DB[name].append(donation)
+        file_name = name + ".txt"
+        try:
+            with open(file_name, "w") as file:
+                file.write(THANK_YOU_EMAIL.format(name, donation))
+        except IOError as e:
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_05/Mail_Room_03.py:124]: OK

File: students/eric_gosnell/lesson_05/Mail_Room_03.py [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+def all_donors_letter():
+    """ Write multiple thank you files and print to screen - uses most recent donation """
+    for donor in DONOR_DB:
+        file_name = donor + ".txt"
+        try:
+            with open(file_name, "w") as file:
+                # Most recent donation
+                file.write(THANK_YOU_EMAIL.format(donor, DONOR_DB[donor][-1]))
+        except IOError as e:
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_05/Mail_Room_03.py:144]: OK

File: students/eric_gosnell/lesson_05/Mail_Room_03.py [hunk 7]
```diff
@@ -0,0 +1,225 @@
+"""
+Eric Gosnell
```
Comment (majstorpoki) [inline students/eric_gosnell/lesson_05/Mail_Room_03.py:2]: Any usage of comprehensions?

**Review by majstorpoki — APPROVED** [review]

**Pull request merged**